zazu/plugins/eslint_styler.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+"""ESLint plugin for zazu."""
+import zazu.styler
+zazu.util.lazy_import(locals(), [
+    'json',
+    'os',
+    'subprocess'
+])
+
+__author__ = "Patrick Moore"
+__copyright__ = "Copyright 2018"
+
+
+class ESLintStyler(zazu.styler.Styler):
+    """ESLint plugin for code styling."""
+
+    def style_string(self, string, filepath):
+        """Fix a string to be within style guidelines.
+
+        Args:
+            string (str): the string to style
+            filepath (str): the filepath of the file being styled
+
+        Returns:
+            Styled string.
+
+        """
+
+        node_modules_path = None
+        eslint = 'eslint'
+        cwd = os.path.normpath(os.getcwd())
+        dirname = os.path.normpath(os.path.dirname(filepath))
+        loop_count = 0
+
+        while True and loop_count < 100:
+            loop_count = loop_count + 1
+            maybe_eslint = os.path.join(dirname, 'node_modules/eslint/bin/eslint.js')
+
+            if os.path.isfile(maybe_eslint):
+                eslint = maybe_eslint
+                break
+            elif dirname == cwd:
+                break
+            elif os.path.realpath(dirname) == '/':
```

Comment:
This check actually seems pretty fragile. I'd replace it and bail out if the a proposed dirname i.e dirname/.. is the same as the old one.  "/.." and "/" should compare equal once you normpath them.